api/src/main/java/org/eclipse/microprofile/config/ConfigProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2016 IBM Corp. and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ *******************************************************************************/
+
+package org.eclipse.microprofile.config;
+
+import javax.enterprise.inject.spi.CDI;
+
+import org.eclipse.microprofile.config.spi.ConfigExtension;
+
+
+/**
+ * <p>
+ * This is the central class to access a {@link Config}.
+ * </p>
+ *
+ *
+ * Example usage:
+ *
+ * <pre>
+ * String restUrl = ConfigProvider.getConfig().getValue(
+ *         &quot;myproject.some.remote.service.url&quot;);
+ * Integer port = ConfigProvider.getConfig().getValue(
+ *         &quot;myproject.some.remote.service.port&quot;, Integer.class);
+ * </pre>
+ *
+ *
+ * @author <a href="mailto:struberg@apache.org">Mark Struberg</a>
+ * @author <a href="mailto:rmannibucau@apache.org">Romain Manni-Bucau</a>
+ * @author <a href="mailto:emijiang@uk.ibm.com">Emily Jiang</a>
+ */
+public final class ConfigProvider {
+    /**
+     * Provide a {@link Config} based on all {@link org.eclipse.microprofile.config.spi.ConfigSource ConfigSources}
+     *
+     * <p>There is exactly a single Config instance per Application</p>
+     * of the current Thread Context ClassLoader (TCCL)
+     *
+     * @return the current {@link Config}
+     */
+    public static Config getConfig() {
```

Comment:
Yes and yes.

1.) Yes, we can actually remove the whole ConfigProvider. 
1.a User code would use @Inject Config (or even more fine granular future features, like the owner-style config you proposed).
1.b Exension developer can @Observes Config or use beanManager.getExtension
1.c Non-CDI parts could still use CDI.current()...

2.) Yes, by mandating CDI in the core parts we loose adoption in every library where CDI is only optional. E.g. Hibernate or OpenJPA will never use it that way. Nor even any other EE component. Because in all of them CDI is only used IF available.
I tried to explain this point for months now, but it seems that people first need to see the mess before they get it (Brexit, Trump, anyone?)